### ISSUE TITLE
Correct the bound of the loss for M = 2

### ISSUE THREAD

**Issue by luiscarlosgph** [luiscarlosgph/jiayuan-huang-phd#45]
```python
import torch.nn.functional as F

# Get raw logits and apply softmax
logits = task_classifier(features)
predictions = F.softmax(logits, dim=1)

# Convert targets to one-hot (assuming 4 classes)
targets_one_hot = F.one_hot(true_labels, num_classes=4).float()

# Use strictly sum-reduced MSE (Brier Score)
classifier_loss = F.mse_loss(predictions, targets_one_hot, reduction='sum')

# Calculate the mean across the BATCH, but keep the SUM across the classes
classifier_loss = classifier_loss / features.size(0)
```

**Comment by PeterHJY628** [comment]
applied
